packages/common/http/src/transfer_cache.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ApplicationRef, inject, ɵmakeStateKey as makeStateKey, ɵStateKey as StateKey, ɵTransferState as TransferState} from '@angular/core';
+import {Observable, of} from 'rxjs';
+import {filter, take, tap} from 'rxjs/operators';
+
+import {HttpHeaders} from './headers';
+import {HttpHandlerFn} from './interceptor';
+import {HttpRequest} from './request';
+import {HttpEvent, HttpResponse} from './response';
+
+interface TransferHttpResponse {
+  body?: any;
+  headers?: Record<string, string[]>;
+  status?: number;
+  statusText?: string;
+  url?: string;
+  responseType?: HttpRequest<unknown>['responseType'];
```

Comment:
nit: since all keys are optional, maybe it would be more descriptive below to use TypeScript's utility types:  `Partial<TransferHttpResponse>` 

```suggestion
  body: any;
  headers: Record<string, string[]>;
  status: number;
  statusText: string;
  url: string;
  responseType: HttpRequest<unknown>['responseType'];
```